sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -655,8 +678,13 @@ def test_check_estimator():
         "Estimator LargeSparseNotSupportedClassifier doesn't seem to "
         r"support \S{3}_64 matrix, and is not failing gracefully.*"
     )
-    with raises(AssertionError, match=msg):
-        check_estimator(LargeSparseNotSupportedClassifier())
+    with ignore_warnings(category=DeprecationWarning):
```

Comment:
I think that I would prefer using `warnings.catch_warnings` and then `warnings.filterwarnings` in which we can configure the error message. It will be more explicit which warning we expect to be raised here.